### PR TITLE
Expose image variant

### LINKED ISSUE #868: Move internal/pkg/platform package
Hi @mtrmac,

I would like to use your [`platform.WantedPlatforms`](https://github.com/containers/image/tree/d85425defc5285493005e476e55a05c1acd69ed5/internal/pkg/platform) function but this is not possible to import internal packages of another package (https://golang.org/doc/go1.4#internalpackages). Would it be possible to outsource it? Something like `github.com/containers/image/pkg/platform` maybe?

Thanks!

### PR DESCRIPTION
**Pull request by crazy-max**
Closes #868

### CODE REVIEW

**Comment by rhatdan** [comment]
@crazy-max you need to sign your PR
git commit -a --amend -s
git push --force

**Comment by crazy-max** [comment]
@rhatdan My commit is already signed:

```
Expose image variant

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
#
# Date:      Sat Mar 28 23:34:53 2020 +0100
#
# On branch master
# Your branch is up to date with 'origin/master'.
#
# Changes to be committed:
#       modified:   manifest/docker_schema2.go
#       modified:   types/types.go
```

image

**Comment by crazy-max** [comment]
@rhatdan Strange, looking at [git-validation DCO regexp](https://github.com/vbatts/git-validation/blob/master/rules/dco/dco.go#L17), that should be ok: https://regex101.com/r/UFZjrs/5

**Comment by mtrmac** [comment]
@crazy-max The commit currently in this PR does not have a `Signed-off-by`.

File: manifest/docker_schema2.go [hunk 1]
```diff
@@ -122,6 +122,8 @@ type Schema2V1Image struct {
 	Architecture string `json:"architecture,omitempty"`
 	// OS is the operating system used to build and run the image
 	OS string `json:"os,omitempty"`
+	// Variant is a variant of the CPU that the image is build and runs on
+	Variant string `json:"variant,omitempty"`
```
Comment (mtrmac) [inline manifest/docker_schema2.go]: In all three places, please move this just below `Architecture`
- That is consistent with docker/docker
- It is a variant of the architecture, not of the OS.

File: manifest/docker_schema2.go [hunk 2]
```diff
@@ -120,6 +120,8 @@ type Schema2V1Image struct {
 	Config *Schema2Config `json:"config,omitempty"`
 	// Architecture is the hardware that the image is build and runs on
 	Architecture string `json:"architecture,omitempty"`
+	// Variant is a variant of the CPU that the image is build and runs on
```
Comment (TomSweeneyRedHat) [inline manifest/docker_schema2.go]: minor nit: build -> built

**Comment by rhatdan** [comment]
LGTM
Thanks @crazy-max

**Pull request merged**